pkg/api/types.go

```diff
@@ -633,6 +635,8 @@ type EnvVar struct {
 type EnvVarSource struct {
 	// Required: Selects a field of the pod; only name and namespace are supported.
 	FieldRef *ObjectFieldSelector `json:"fieldRef"`
+	// Required: Selects a field of the pod; only name and namespace are supported.
+	ConfigDataRef *ConfigDataSelector `json:"configDataRef"`
```

Comment:
+1 to clayton's comments.  The sticky part is that we'll have to define what the rules for selecting fields from `ConfigData` are with `FieldPath`.